assembly/src/assembler/module_provider.rs

```diff
@@ -0,0 +1,59 @@
+use super::{BTreeMap, Library, LibraryError, Module, ProcedureId, Vec};
+
+// MODULE PROVIDER
+// ================================================================================================
+
+/// A module provider that will store the procedure ASTs used during the compilation process.
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct ModuleProvider {
+    modules: Vec<Module>,
+    /// Map from procedure id to module index + proc index in the module.
```

Comment:
nit: I think this should be more like: "Map from procedure id to the index of a module in which the procedure is defined."

Reply:
Fixed!